Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiManager.cpp

```diff
@@ -331,23 +332,23 @@ namespace AzToolsFramework::ViewportUi
 
     ClusterId ViewportUiManager::RegisterNewCluster(AZStd::shared_ptr<Internal::ButtonGroup>& buttonGroup)
     {
-        ClusterId newId = ClusterId(m_clusterButtonGroups.size() + 1);
+        ClusterId newId = ClusterId(AZ::Sfmt::GetInstance().Rand64());
```

Comment:
Hmm I'm not sure just using a random number here is the way to go 🤔 I know it's very unlikely but what happens if/when there is a collision? Is it because we should be using a different size value (should it be buttonGroups.size() + number of buttons per group.size() instead?) Or we should use a separate monotonically increasing counter to give a unique id and not rely on the size value at all, that might be the simplest option.

Reply:
I think a counter should be find. exhausting a u64 would be a challenge.

Reply:
Nice, yeah let's go with that! We'd have to wait for the universe to end multiple times with a u64 😅 Thanks!